internal/parser/reader_parser.go

```diff
@@ -22,6 +22,12 @@ var splitFunc bufio.SplitFunc = func(data []byte, _ bool) (advance int, token []
 		}
 	}
 
+	if advance == len(data) && !atEOF {
+		// We have reached the end of the buffer but have not yet seen two consecutive
+		// newline sequences, so we request more data.
+		return 0, nil, nil
+	}
+
 	if l := len(data); advance < l {
 		advance++
 		if advance < l && data[advance-1] == '\r' && data[advance] == '\n' {
```

Comment:
Can we write it like this maybe, so we don't repeat the `len(data)`? It's also a good opportunity to document some surrounding code.
```suggestion
	if l := len(data); advance == l && !atEOF {
		// We have reached the end of the buffer but have not yet seen two consecutive
		// newline sequences, so we request more data.
		return 0, nil, nil
	} else if advance < l {
		// We have found a newline. Consume the end-of-line sequence.
		advance++
		// Consume one more character if end-of-line is \r\n.
		if advance < l && data[advance-1] == '\r' && data[advance] == '\n' {
```

Reply:
Done. I also added a comment at the top of `splitFunc` to try and describe what it does at a high level, since I had a bit of a hard time figuring it out the first time around.